network/identity.go

```diff
@@ -117,13 +128,37 @@ func (i *identity) start() error {
 	return nil
 }
 
-func (i *identity) getStatus() *proto.Status {
-	return &proto.Status{
-		Chain: int64(i.srv.config.Chain.Params.ChainID),
+func (i *identity) getStatus(peerID peer.ID) *proto.Status {
+	status := &proto.Status{
+		Metadata: make(map[string]string, 1),
+		Chain:    int64(i.srv.config.Chain.Params.ChainID),
+	}
+	if _, ok := i.srv.temporaryDials.Load(peerID); ok {
+		status.TemporaryDial = true
+	}
+
+	return status
+}
+
+// checkSlots checks for the available connection slots and disconnects if slots are full
+func (i *identity) checkSlots(direction network.Direction, peerID peer.ID) (slotsFull bool) {
```

Comment:
`checkSlots` would confuse because you will disconnect from peer inside, checking is not the main task here.

Reply:
Renamed it, Commit [link](https://github.com/0xPolygon/polygon-edge/pull/343/commits/23abbb4760584a06d5fbabd156a6bb83eef8b0ee)